src/lib/components/Button.tsx

```diff
@@ -14,28 +14,46 @@ export const BaseButton = styled.button`
   line-height: inherit;
   margin: 0;
   padding: 0;
+  transition: filter 0.125s linear;
 
   :disabled {
     cursor: initial;
     filter: saturate(0) opacity(0.4);
+    transition: none;
```

Comment:
will this stop transitions out of a disabled state?, and is [filter](https://developer.mozilla.org/en-US/docs/Web/CSS/filter) the only property you want to animate?

Reply:
Should the border and background colors below also use this transition?

Reply:
That's much simpler, yes. I've updated the Button to use transition instead of animation.